pkg/devfile/parser/data/2.0.0/components.go

```diff
@@ -11,6 +12,33 @@ func (d *Devfile200) GetComponents() []common.DevfileComponent {
 	return d.Components
 }
 
+// AddComponents adds the slice of DevfileComponent objects to the devfile's components
+// if a component is already defined, error out
+func (d *Devfile200) AddComponents(components []common.DevfileComponent) error {
+	componentMap := make(map[string]bool)
+	for _, component := range d.Components {
+		componentMap[component.Container.Name] = true
+	}
+
+	for _, component := range components {
+		if _, ok := componentMap[component.Container.Name]; !ok {
+			d.Components = append(d.Components, component)
+		} else {
+			return fmt.Errorf("component %v is already present in the devfile", component.Container.Name)
```

Comment:
can we make a new error type here like `AlreadyPresent`? and return this.
and the place where you are calling `AddComponents`, checks for this type of error and clearly prints
`This component is already present in parent devfile, to override it define as parent field`

Reply:
> and the place where you are calling AddComponents, checks for this type of error and clearly prints

We can do that. But this is only line in the function which throws an error. The only error this function can throw is also due to the component being already present. So it would be unnecessary to check the error type at the caller i.e AddComponents().

Reply:
sure, as you wish

Reply:
How about instead of error wrapping, AddComponents() prints the message directly?

Reply:
OK let me check.